### PR TITLE
feat: tweak bitcoin queries for extension

### PR DESCRIPTION
**Pull request by edgarkhanzadian**
Connected to this PR in extension: https://github.com/leather-wallet/extension/pull/5343

This one got final touches for the Great Bitcoin Queries Move. Tested on extension using pnpm hardlinking, works fine.

As soon as this one is merged and released, i'll open the extension's PR for review

### CODE REVIEW

File: packages/query/config/wallet-config.json [hunk 1]
```diff
@@ -0,0 +1,105 @@
+{
```
Comment (pete-watters) [inline packages/query/config/wallet-config.json:1]: This seems to be more general settings. Does it belong inside of `query` ?

File: packages/query/src/bitcoin/address/transactions-by-address.query.ts [hunk 2]
```diff
@@ -17,12 +17,13 @@ export function useGetBitcoinTransactionsByAddressQuery<T extends unknown = Bitc
   return useQuery({
     enabled: !!address,
     queryKey: ['btc-txs-by-address', address],
-    queryFn: () => client.addressApi.getTransactionsByAddress(address),
+    queryFn: async ({ signal }) => {
+      return client.addressApi.getTransactionsByAddress(address, signal);
```
Comment (pete-watters) [inline packages/query/src/bitcoin/address/transactions-by-address.query.ts:21]: Maybe we can retain the implicit return?

File: packages/query/src/bitcoin/ordinals/brc20/brc20-tokens.hooks.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  useGetBrc20TokensQuery,
+  useLeatherNetwork,
+} from '@leather-wallet/query';
+import { createMoney, unitToFractionalUnit } from '@leather-wallet/utils';
+import { P2TROut } from '@scure/btc-signer';
+import BigNumber from 'bignumber.js';
+
+import type { Brc20Token } from '../../bitcoin-client';
+
+// ts-unused-exports:disable-next-line
```
Comment (pete-watters) [inline packages/query/src/bitcoin/ordinals/brc20/brc20-tokens.hooks.ts:15]: Maybe we can ditch some of these unused exports?

File: packages/query/src/bitcoin/ordinals/brc20/brc20-tokens.query.ts [hunk 4]
```diff
@@ -105,17 +105,16 @@ export function useGetBrc20TokensQuery({
...
-  //   // }
-  //   // eslint-disable-next-line react-hooks/exhaustive-deps
-  // }, [query.hasNextPage]);
+  useEffect(() => {
+    const brc20AcrossAddressesCount = query.data?.pages.reduce((acc, page) => {
+      return acc + page.brc20Tokens.flatMap(item => item).length;
+    }, 0);
+
+    if (!query.hasNextPage && brc20AcrossAddressesCount && brc20AcrossAddressesCount > 0) {
+      // void analytics.identify({
```
Comment (pete-watters) [inline packages/query/src/bitcoin/ordinals/brc20/brc20-tokens.query.ts:114]: Maybe we can remove `analytics` here?

File: packages/query/src/bitcoin/transaction/use-bitcoin-broadcast-transaction.ts [hunk 5]
```diff
@@ -19,9 +19,8 @@ interface BroadcastCallbackArgs {
 export function useBitcoinBroadcastTransaction() {
   const client = useBitcoinClient();
   const [isBroadcasting, setIsBroadcasting] = useState(false);
-  // TODO: analytics should be handled on app level
```
Comment (pete-watters) [inline packages/query/src/bitcoin/transaction/use-bitcoin-broadcast-transaction.ts:22]: Can we get rid of the commented analytics code also?

```
// const analytics = useAnalytics();
```

**Review by pete-watters — APPROVED** [review]
Nice work 👍 I added a few suggestions but nothing  blocking

File: packages/query/config/wallet-config.json (hunk 1 above)
Comment (kyranjamie) [reply packages/query/config/wallet-config.json:1]: +1 would not put the config here, but we could still have the query for it?

**Review by fbwoolf — APPROVED** [review]

**Review by kyranjamie — APPROVED** [review]

**Pull request merged**